frontend/src/pages/choose-role/choose-role.js

```diff
@@ -0,0 +1,70 @@
+import { ENDPOINTS } from '../../api/config';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import Form from '../../components/input-form/input-form';
+import Button from '../../components/button/button';
+import ToggleButtonGroup from '../../components/toggle-button-group/toggle-button-group';
+import './choose-role.css';
+
+
+const ChooseRole = () => {
+
+    const [selectedRole, setSelectedRole] = useState(1); // Default to "Startup" (1)
+    const navigate = useNavigate();
+    const url = `${ENDPOINTS.CHANGE_ROLE}`;
+
+    const handleConfirm = async () => {
+        const accessToken = localStorage.getItem('access_token'); // Get token from localStorage
+        if (!accessToken) {
+          console.error('No access token found');
+          navigate('/'); // Redirect to home if no token
+          return;
+        }
+    
+        try {
+          const response = await fetch(url, {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: `Bearer ${accessToken}`, // Include drf-social-oauth2 token in Authorization header
+            },
+            body: JSON.stringify({ role: selectedRole }),
+          });
+    
+          if (!response.ok) {
+            throw new Error('Failed to change role');
+          }
+          
+          const data = await response.json();
+
+          // Save JWT tokens in localStorage
+          localStorage.setItem('accessToken', data.access);
+          localStorage.setItem('refreshToken', data.refresh);
+
+          // Remove drf-social-oauth2 tokens from localStorage
+          localStorage.removeItem('access_token');
+          localStorage.removeItem('refresh_token');
+
+          // Redirect to profile page on success
+          navigate('/my_profile/');
+        } catch (error) {
+          console.error(error.message);
+          // Redirect to home page on error
+          navigate('/');
+        }
+      };
```

Comment:
Disable the "Confirm" button while the API call is in progress to prevent multiple submissions.
```
const [isSubmitting, setIsSubmitting] = useState(false);

const handleConfirm = async () => {
    if (isSubmitting) return;
    setIsSubmitting(true);

    try {
        // Existing logic...
    } catch (error) {
        console.error(error.message);
    } finally {
        setIsSubmitting(false);
    }
};

return (
    <Button
        text="Confirm"
        variant="primary"
        disabled={isSubmitting}
        onClick={(e) => { e.preventDefault(); handleConfirm(); }}
    />
);
```